CONTRIBUTING.md

```diff
@@ -0,0 +1,146 @@
+# CONTRIBUTING
+
+## Introduction
+
+Hello and thank you for your interest in contributing to Log Viewer.
+
+Contributions are welcome and there are many ways you can get involved!
+
+To get started, choose your area of interest:
+
+<table>
+  <tr>
+    <td  align="center">
+        <a href="#-issues--discussions">👥 Issues & Discussions</a> |
+        <a href="#-documentation">📚 Documentation</a> |
+        <a href="#-spread-the-word">📣 Spread the word</a> |
+        <a href="#-code-contribution">💻 Code Contribution</a>
+    </td>
+  </tr>
+</table>
+
+<br/>
+
+---
+
+### 👥 Issues & Discussions
+
+You can interact with users by sharing information and asking/answering questions in our [Discussions](https://github.com/opcodesio/log-viewer/discussions) tab.
+
+Also, you can contribute by reporting bugs, patching problems or providing technical support in our [Issues](https://github.com/opcodesio/log-viewer/issues) tab.
+
+<br/>
+
+---
+
+### 📚 Documentation
+
+Documentation is key for any project success!
+
+Currently, our documentation is stored at the [README](https://github.com/opcodesio/log-viewer/blob/main/README.md) file of this repository.
+
+You may contribute by improving existing information, covering missing topics, or fixing typos and grammar errors.
+
+The documentation official language is in English.
+
+<br/>
+
+---
+
+### 📣 Spread the word
+
+If you enjoy Log Viewer, please consider talking about our project in your community.
+
+Share this [repository link](https://github.com/opcodesio/log-viewer/) on Twitter, YouTube, Discord or any other social network you are part of.
+
+You are also welcome to write articles, reviews and tutorials about this project on your blog or programming website.
+
+Ah! Don't forget to let the author know about your work. Say hello to [@arukompas](https://github.com/arukompas).
+
+<br/>
+
+---
+
+### 💻 Code Contribution
+
+Please follow the steps below to contribute with code.
+
+## Steps
+
+### 📌 Step 1
+
+Fork this repository and enter its directory. Run the command:
+
+```shell
+git clone https://github.com/opcodesio/log-viewer.git && cd log-viewer
+```
+
+### 📌 Step 2
+
+Install all PHP dependencies using Composer, run the command:
+
+```shell
+composer install
+```
+
+Once finished, proceed to install Node dependencies. Run the command:
+
+```shell
+npm install
+```
+
+### 📌 Step 3
+
+Create a new branch for your code. You may call it `feature-` / `fix-` / `enhancement-` followed by the name of what you are developing.
+
+For example:
+
+```shell
+git checkout -b feature/feature-new_about_page
+```
+
+Please consider to write tests to cover your code. Tests are helpful for the code reviewers, and it reduces bugs and improves software quality.
+
```

Comment:
What about introducing a step between step 3 (git checkout) and step 4 (tests before committing).

It would be a few tips about the actual development, such as:
1. when developing a Laravel package, you may want to use the local package code inside your Laravel application. Here's how to set it up - (instructions on setting up the composer.json file with a custom "path" repository that points to the log-viewer package on the machine)
2. when developing, run `npm run watch` so that your frontend changes are re-built automatically
3. if you've made any frontend changes, make sure to build the assets for production before committing, by using `npm run production`